src/js/elements/main-nav.ts

```diff
@@ -37,41 +37,42 @@ export class MainNav {
             listItem.toggleEl.addEventListener('click', () => {
                 this._toggleListItemState(listItem);
             })
-            listItem.addEventListener('keyup', (event: KeyboardEvent) => {
-                // @ts-ignore
-                this._handleKeyUp(event, event.target);
-            });
+            listItem.addEventListener('keyup', this._handleKeyUp);
         })
     }
 
     /**
-     * - On Enter or Space click on a main nav chevron we toggle the nav item open / closed
-     * - On Esc press close the open main navigation
+     * Close parent or all open navigation lists
      * @param   event
-     * @param   listItem
      */
-    _handleKeyUp = (event: KeyboardEvent, listItem: MainNavListItem) => {
-        // Open list item when there is "enter" or "space" press on the item chevron
-        if (event.key === 'Enter' || event.code === 'Enter' || event.code === 'Space' || event.key === ' ') {
-            if (listItem && listItem.classList.contains(this.listItemChevronSelector)) {
-                const parentListItem = listItem.closest(this.listItemSelector) as MainNavListItem;
-                this._toggleListItemState(parentListItem);
-            }
-        }
+    _handleKeyUp = (event: KeyboardEvent) => {
+        /* @ts-ignore */
+        const parentListItem = event.target?.offsetParent;
 
-        // Close all nav items when "Escape" is pressed
         if (event.key === 'Escape' || event.code === 'Escape') {
-            this._closeAllNavListItems();
+            // Close parent nav item if parent listItem is open
+            if (parentListItem?.hasAttribute('is-open')) {
+                event.stopPropagation();
+                this._toggleListItemState(parentListItem);
+            }
+            // Close all nav items when "Escape" is pressed
+            else this._closeAllNavListItems();
```

Comment:
```suggestion
            else {
                this._closeAllNavListItems();
            }    
```
I would prefer brackets for the else. What do you think @luis-pato ? Not sure if we have styleguide/linting in this repo.
And the comments can be removed for all I care. The code is verbose enough

Reply:
The comments will be stripped anyway when compiling TS to JS. While I agree that they're not strictly neccessary, I don't see any harm in keeping them in.
Regarding the brackets, I'd trust @luis-pato's judgement :)

Reply:
Yes, of course it doesn't hurt, but perhaps it's redundant because there is already a comment for the function. `Close parent or all open navigation lists`
But it doesn't really matter

Reply:
I also prefer having brackets for the else block